walk-in-interview/src/main/java/com/google/job/servlets/JobsListingsServlet.java

```diff
@@ -0,0 +1,221 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+import java.sql.Time;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.*;
+
+/**
+ * Servlet that handles getting the job listings given the filters.
+ */
+@WebServlet("/jobs/listings")
+public final class JobsListingsServlet extends HttpServlet {
+    private static final long TIMEOUT = 5;
+
+    public static final String MIN_LIMIT_PARAM = "minLimit";
+    public static final String MAX_LIMIT_PARAM = "maxLimit";
+    public static final String REGION_PARAM = "region";
+    public static final String SORT_BY_PARAM = "sortBy";
+    public static final String ORDER_PARAM = "order";
+    public static final String PAGE_SIZE_PARAM = "pageSize";
+    public static final String PAGE_INDEX_PARAM = "pageIndex";
+
+    private JobsDatabase jobsDatabase;
+
+    @Override
+    public void init() {
+        this.jobsDatabase = new JobsDatabase();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        try {
+            int minLimit = parseMinLimit(request);
+            int maxLimit = parseMaxLimit(request);
+            SingaporeRegion region = parseRegion(request);
+            Filter sortBy = parseSortBy(request);
+            Order order = parseOrder(request);
+            int pageSize = parsePageSize(request);
+            int pageIndex = parsePageIndex(request);
+
+            JobPage jobPage = fetchJobPageDetails(minLimit, maxLimit, region, sortBy, order,
+                pageSize, pageIndex);
+
+            String json = new Gson().toJson(jobPage);
+            response.setContentType("application/json;");
+            response.getWriter().println(json);
+        } catch(IllegalArgumentException | ServletException | ExecutionException | TimeoutException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    /**
+     * Returns the JobPage object.
+     *
+     * @param minLimit The lower limit to be shown (inclusive).
+     * @param maxLimit The upper limit to be shown (inclusive).
+     * @param region The region in Singapore.
+     * @param sortBy The sorting of the list of jobs.
+     * @param order The ordering of the sorting.
+     * @param pageSize The number of job listings to be returned.
+     * @param pageIndex The page which we are on (pagination).
+     * @return JobPage object with all the details for the GET response.
+     */
+    private JobPage fetchJobPageDetails(int minSalary, int maxSalary, SingaporeRegion region,
+        Filter sortBy, Order order, int pageSize, int pageIndex) throws ServletException, ExecutionException, TimeoutException {
+        try {
+            return this.jobsDatabase.fetchJobPage(minSalary, maxSalary, region, sortBy, order, pageSize, pageIndex)
```

Comment:
now that you have 7 arguments in there, you might want to consider passing a QueryBuilder object with default sensible values for order, pageSize, pageIndex.